src/sentry/snuba/metrics/extraction.py

```diff
@@ -1268,8 +1279,11 @@ def to_metric_spec(self, project: Project) -> MetricSpec:
         tag_from_groupbys = self.tags_groupbys(self.groupbys)
         extended_tags_conditions.extend(tag_from_groupbys)
 
+        # Once we switch to the next spec we can remove this block
+        # since the environment will be added to the groupbys, thus, being included in the query hash
         if (
             self.spec_type == MetricSpecType.DYNAMIC_QUERY
+            and self.spec_version.flags == set()
```

Comment:
Not blocking; I don't know that we'll need this flag matching, it's pretty complicated, I can't see us needing a lot of individual behaviours and every new ability to a spec version will essentially reset the spec anyway.